File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/FragmentDelegate.java
package me.ameriod.lib.mvp.view.deligate;

import android.os.Bundle;
import android.support.annotation.Nullable;

import me.ameriod.lib.mvp.Mvp;

public interface FragmentDelegate<V extends Mvp.View, P extends Mvp.Presenter<V>> {

    void onViewCreated(@Nullable Bundle savedInstanceState);

    void onSaveInstanceState(Bundle outState);

    void onDestroyView();

    P getPresenter();
}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/FragmentDelegate.kt
package me.ameriod.lib.mvp.deligate

import android.os.Bundle

import me.ameriod.lib.mvp.Mvp

interface FragmentDelegate<in V : Mvp.View, out P : Mvp.Presenter<in V>> {

    fun onViewCreated(savedInstanceState: Bundle?)

    fun onSaveInstanceState(outState: Bundle)

    fun onDestroyView()

    fun getPresenter(): P
}
